### ISSUE TITLE
FieldArrayType different valueChange behavior add/remove - v6

### ISSUE THREAD

**Issue by FritzHerbers** [ngx-formly/ngx-formly#3353]
When adding an array object, in the valueChange Observable the model and the fieldGroup length are the same.
When removing an array object, the model size is correct, the fieldGroup length hasn't been changed. Is this a bug or what is the reason?

https://stackblitz.com/edit/angular-aty7mb-rwwl94?file=src%2Fapp%2Frepeat-section.type.ts

Klick 'Add'
Klic 'Remove'
Check the console log for the lengths of the model and the fieldGroup.

ngx-formly: 6.0.0-beta.3

**Comment by ajojra** [comment]
just needs next digestion cycle looks like: 

https://stackblitz.com/edit/angular-aty7mb-a6pfap?file=src/app/repeat-section.type.ts

**Comment by FritzHerbers** [comment]
>just needs next digestion cycle looks like:

I was aware of that.

```
  add(i?: number, initialModel?: any, { markAsDirty } = { markAsDirty: true }) {
    i = i == null ? this.field.fieldGroup.length : i;
    if (!this.model) {
      assignFieldValue(this.field, []);
    }

    this.model.splice(i, 0, initialModel ? clone(initialModel) : undefined);
    this._build();
    markAsDirty && this.formControl.markAsDirty();
  }

  remove(i: number, { markAsDirty } = { markAsDirty: true }) {
    this.model.splice(i, 1);
    unregisterControl(this.field.fieldGroup[i], true);
    this.field.fieldGroup.splice(i, 1);
    this.field.fieldGroup.forEach((f, key) => (f.key = `${key}`));
    this._build();
    markAsDirty && this.formControl.markAsDirty();
  }
```

With remove, I found out that the `valueChanges` happen during `unregisterControl`, and not what I thought during `_build()` (as with `add()`, so that the `fieldGroup` gets spliced later.

With add/remove, there is no `emit` flag to have the specialized array type decide itself for `valueChanges`. 
In our case, when the last row is deleted, a default row can be inserted, which generates two `valueChanges`. Elsewhere this generates unwanted events, also the model logically is incorrect.

@aitboudad 
Would it make sense to have an `emit` flag with add/remove. Thereafter the specialized array type can set the model (or other mechanism provided by formly) to emit the valueChange itself ?

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.0.0-rc.0](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.0.0-rc.0) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
